monai/data/image_reader.py

```diff
@@ -418,6 +423,7 @@ def __init__(
         swap_ij: bool = True,
         prune_metadata: bool = True,
         label_dict: dict | None = None,
+        fname_regex: str = r"^(?!.*LICENSE).*",
```

Comment:
Is this excluding enough? What about files starting with "." which are meant to be hidden, anything with README in the title, DICOMDIR files, or other known file types? Was there a specific reason why LICENSE was important to skip? 

I would also put this default value into a constant at the top of this file to make it easier to spot and reuse elsewhere.

Reply:
good question, this is mainly to be consistent with the existing logic https://github.com/Project-MONAI/MONAI/blob/3239db8152cad7e3997a0e09147863c9b34be14b/monai/data/image_reader.py#L470-L471 @yiheng-wang-nv are these from the tcia dataset?

Reply:
Perhaps we could directly set the rules for inclusion, something like r".*.dcm|.*.DCM"?

Reply:
Our test cases have no extension names.. https://github.com/Project-MONAI/MONAI/tree/dev/tests/testing_data/CT_DICOM

Reply:
>@yiheng-wang-nv are these from the tcia dataset?
Hi @wyli , yes, for TCIA dataset, a LICENSE file will be included in the folder, thus I wrote this logic.

Reply:
Ok thanks, i'll move the filter to monai.apps.datasets.